### ISSUE TITLE
Preloader boilerplate templates will omit pipe from renamed conditional links

### ISSUE THREAD

**Issue by seankurth** [Mediawiki-wysiwyg/WYSIWYG-CKeditor#30]
When a template is added from preloader, a link that says, for example,

```
[[Help:{{ROOTPAGENAME}}/Cats|Taking care of {{ROOTPAGENAME}}'s cats]]
```

Should say "Taking care of Doe, Jane's cats." And in the template, that's exactly what it does. But preload that page into WYSIWYG, and what you get is:

```
[[Help:{{ROOTPAGENAME}}/Cats]]
```

The result is a link that says "Help:Doe, Jane/Cats." Not nearly as pretty. And that's just on the page, it gets worse in the WYSIWYG editor window, which will just say "Help:Clients/July/Doe, Jane/Cats" rather than collecting the root page name.

**Comment by riparap** [comment]
MW variables were enabled for some reason only with external links. Also some variables and magic words of latest MW versions were missing in wysiwyg. 

I have made modifications and, at least with your example, it does not corrupt structure of [[ ...{{ROOTPAGENAME }}...| ... {{ROOTPAGENAME}}...]] -definition anymore, so I guess it should work better now. This fix may have effects on other links too, I hope it works, because my tests were limited to few simple cases.

In wysiwyg it will not show the final text if you use variables in text which is displayed on page, you will see name of variable instead within the sting of text. Only after you save the page and exit wysiwyg, final value of variable is displayed on page.

**Comment by seankurth** [comment]
OK thanks, WYSIWYG is close to perfect now

On 11/27/14, riparap notifications@github.com wrote:

> MW variables were enabled for some reason only with external links. Also
> some variables and magic words of latest MW versions were missing in
> wysiwyg. I have made modifications and at least with your example it keeps
> structure of [[ ...{{ROOTPAGENAME }}...| ... {{ROOTPAGENAME}}...]]
> -definition intact, so I guess it should work better now.
> 
> In wysiwyg it will not show the final text if you use variables in text
> which is displayed on page, you will see name variable instead. When you
> save the page and exit wysiwyg, value of variable is visible on page.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/Mediawiki-wysiwyg/WYSIWYG-CKeditor/issues/30#issuecomment-64783227

## 

_Sean Kurth_

**Comment by riparap** [comment]
Fixed in version at "26.11.2014" .
